packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -68,12 +69,22 @@ class RunCommand extends RunCommandBase {
         defaultsTo: false,
         negatable: false,
         help: 'Start in a paused mode and wait for a debugger to connect.');
+    argParser.addFlag('pub',
```

Comment:
I'm wondering if we could get away w/o a flag, since we only shell out to pub if the pubspec is our-of-date.